.github/workflows/github-actions-ci.yml

```diff
@@ -22,30 +21,34 @@ jobs:
           echo "List of changed files:"
           echo ${CHANGED_PYFILES}
           flake8 ${CHANGED_PYFILES} --count
-
+  
   ecephys_etl_pipelines:
-    name: ${{ matrix.python-version }}, ${{ matrix.os }}
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        os: ["macos-latest", "windows-latest", "ubuntu-latest"]
-        python-version: ["3.6", "3.7"]
-    defaults:
-      run:
-        shell: bash -l {0}
+    runs-on: "ubuntu-latest"
     steps:
-      - uses: actions/checkout@v2
-      - uses: conda-incubator/setup-miniconda@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
         with:
-          auto-update-conda: true
-          python-version: ${{ matrix.python-version }}
-          activate-environment: test-env
-      - name: Install
+          username: alleninstitutepika
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build Docker Image
         run: |
-          conda activate test-env
-          pip install codecov
-          pip install -r test_requirements.txt
-          pip install .
-      - name: Test
+          git clone https://github.com/AllenInstitute/ecephys_etl_pipelines.git
+          cd ecephys_etl_pipelines
+          echo "Using branch $GITHUB_HEAD_REF"
+          git checkout $GITHUB_HEAD_REF
+          docker build --build-arg ECEPHYS_ETL_TAG=${GITHUB_HEAD_REF} --build-arg ECEPHYS_ETL_COMMIT_SHA=${GITHUB_SHA} --tag alleninstitutepika/ecephys_etl_pipelines:$GITHUB_SHA docker/
+      - name: Run Tests in Docker Image
         run: |
-          py.test --cov=ecephys_etl_pipelines -n 4
+          docker run alleninstitutepika/ecephys_etl_pipelines:$GITHUB_SHA pytest .
```

Comment:
`--read-only` is important:
https://github.com/AllenInstitute/ophys_etl_pipelines/blob/b73f30e044a39a00d075b297fa858eece14242b5/.circleci/config.yml#L32
https://sylabs.io/guides/3.7/user-guide/singularity_and_docker.html#best-practices